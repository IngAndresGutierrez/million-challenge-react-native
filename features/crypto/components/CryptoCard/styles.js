import { StyleSheet } from 'react-native'

export const styles = StyleSheet.create({
  card: {
    marginBottom: 16,
    borderRadius: 12,
    backgroundColor: '#fff',
  },
  price: {
    marginBottom: 8,
  },
})
