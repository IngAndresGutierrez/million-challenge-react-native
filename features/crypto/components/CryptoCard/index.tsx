import React from 'react'
import { Card, Text, useTheme } from 'react-native-paper'

import { styles } from './styles'
import { CryptoCardProps } from '../../types'

const CryptoCard = ({ coin, onPress }: CryptoCardProps) => {
  const theme = useTheme()

  return (
    <Card style={styles.card} mode="elevated" onPress={onPress} elevation={1}>
      <Card.Content>
        <Text variant="headlineSmall">
          {coin.name} ({coin.symbol})
        </Text>

        <Text
          variant="titleMedium"
          style={[styles.price, { color: theme.colors.primary }]}
        >
          USD ${parseFloat(coin.price_usd).toFixed(2)}
        </Text>

        <Text>Rank: {coin.rank}</Text>
        <Text>
          Market Cap: ${parseFloat(coin.market_cap_usd).toLocaleString('es-ES')}
        </Text>
        <Text>Supply: {parseFloat(coin.csupply).toLocaleString('es-ES')}</Text>
        <Text>Change 24h: {coin.percent_change_24h}%</Text>
      </Card.Content>
    </Card>
  )
}

export default CryptoCard
