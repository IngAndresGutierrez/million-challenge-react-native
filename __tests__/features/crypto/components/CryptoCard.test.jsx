import React from 'react'
import { render } from '@testing-library/react-native'
import { Provider as PaperProvider } from 'react-native-paper'
import CryptoCard from '../../../../features/crypto/components/CryptoCard'

const mockCoin = {
  id: '1',
  name: 'Bitcoin',
  symbol: 'BTC',
  price_usd: '45000',
  rank: '1',
  market_cap_usd: '850000000000',
  csupply: '19000000',
  volume24: '35000000000',
  percent_change_24h: '2.5',
  nameid: 'bitcoin',
  percent_change_1h: '0.5',
  percent_change_7d: '5.0',
  volume24a: 30000000000,
  tsupply: '21000000',
  msupply: '21000000',
  price_btc: '1.0',
}

describe('CryptoCard', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <PaperProvider>
        <CryptoCard coin={mockCoin} onPress={() => {}} />
      </PaperProvider>
    )

    expect(toJSON()).toMatchSnapshot()
  })
})
