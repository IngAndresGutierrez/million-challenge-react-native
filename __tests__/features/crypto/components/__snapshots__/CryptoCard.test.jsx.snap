// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoCard renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#fff",
          "borderRadius": 12,
          "marginBottom": 16,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 3,
        }
      }
      testID="card-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#fff",
            "borderRadius": 12,
            "elevation": 1,
            "flex": undefined,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 1,
          }
        }
        testID="card-container"
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              [
                {
                  "flexShrink": 1,
                },
                undefined,
              ]
            }
            testID="card"
          >
            <View
              borderRadiusStyles={
                {
                  "borderRadius": 12,
                }
              }
              style={
                [
                  {
                    "paddingHorizontal": 16,
                  },
                  {
                    "paddingVertical": 16,
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "fontFamily": "System",
                        "fontSize": 24,
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 32,
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Bitcoin
                 (
                BTC
                )
              </Text>
              <Text
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": "500",
                        "letterSpacing": 0.15,
                        "lineHeight": 24,
                      },
                      [
                        {
                          "marginBottom": 8,
                        },
                        {
                          "color": "rgba(103, 80, 164, 1)",
                        },
                      ],
                    ],
                  ]
                }
              >
                USD $
                45000.00
              </Text>
              <Text
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    {
                      "writingDirection": "ltr",
                    },
                    undefined,
                  ]
                }
              >
                Rank: 
                1
              </Text>
              <Text
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    {
                      "writingDirection": "ltr",
                    },
                    undefined,
                  ]
                }
              >
                Market Cap: $
                850.000.000.000
              </Text>
              <Text
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    {
                      "writingDirection": "ltr",
                    },
                    undefined,
                  ]
                }
              >
                Supply: 
                19.000.000
              </Text>
              <Text
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    {
                      "writingDirection": "ltr",
                    },
                    undefined,
                  ]
                }
              >
                Change 24h: 
                2.5
                %
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
